libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -628,7 +631,7 @@ public OffRoute getOffRouteEngine() {
   public void onProgressChange(Location location, RouteProgress routeProgress) {
     Timber.v("Arrived event occurred");
     sessionState = sessionState.toBuilder().arrivalTimestamp(new Date()).build();
-    NavigationMetricsWrapper.arriveEvent(sessionState, routeProgress, location);
+    NavigationMetricsWrapper.arriveEvent(sessionState, routeProgress, location, locationEngineName);
```

Comment:
@cammace would it make sense to move the arrive event to `NavigationService` like the rest of the events? That way most of the current changes to this class could be self contained within `NavigationService`. @Guardiola31337

Reply:
Don't immediately see any issue with moving the entire `onProgressChange` into NavigationService` to keep all the events in the same class. Though I think whichever path we take, we should only have 1 internal onProgressChange to keep things simple. Let's do this in a different PR and go ahead and merge this one?

Reply:
@cammace @ericrwolfe 
Follow up ticket 👉 https://github.com/mapbox/mapbox-navigation-android/issues/402